samples/net/sockets/echo_client/src/echo-client.c

```diff
@@ -276,7 +276,10 @@ static void init_app(void)
 	}
 
 	init_vlan();
-	init_udp();
+
+	if (IS_ENABLED(CONFIG_NET_UDP)) {
+		init_udp();
+	}
```

Comment:
Could we do it same way as how `init_vlan()` is "guarded", so having a dummy init function in `common.h`